src/Formats/FormatSettings.h

```diff
@@ -141,29 +76,89 @@ struct FormatSettings
         std::string row_between_delimiter;
         std::string field_delimiter;
         std::string escaping_rule;
-    };
+    } custom;
 
-    Custom custom;
+    struct
+    {
+        bool quote_64bit_integers = true;
+        bool quote_denormals = true;
+        bool escape_forward_slashes = true;
+        bool serialize_as_strings = false;
+    } json;
 
-    struct Avro
+    struct
     {
-        String schema_registry_url;
-        String output_codec;
-        UInt64 output_sync_interval = 16 * 1024;
-        bool allow_missing_fields = false;
-    };
+        UInt64 row_group_size = 1000000;
+    } parquet;
+
+    struct Pretty
+    {
+        UInt64 max_rows = 10000;
+        UInt64 max_column_pad_width = 250;
+        UInt64 max_value_width = 10000;
+        bool color = true;
+
+        bool output_format_pretty_row_numbers = false;
+
+        enum class Charset
+        {
+            UTF8,
+            ASCII,
+        };
 
-    Avro avro;
+        Charset charset = Charset::UTF8;
+    } pretty;
 
-    struct Regexp
+    struct
+    {
+        bool write_row_delimiters = true;
+        /**
+         * Some buffers (kafka / rabbit) split the rows internally using callback
+         * so we can push there formats without framing / delimiters (like
+         * ProtobufSingle). In other cases you can't write more than single row
+         * in unframed format.
+         * Not sure we need this parameter at all, it only serves as an additional
+         * safety check in ProtobufSingle format, but exporting constant-size
+         * records w/o delimiters might be generally useful, not only for Kafka.
```

Comment:
If we will allow that - users will successfully create a binary file which no parser can read. And will think that we are producing incorrect Protobuf unable to understand what is wrong with that file (it's binary).

Fixed-length protobuf is something like fixed-length XML  Theoretically possible, but practically useless. 

And same as with XML when you glue the XML the result can also have some (different) meaning. 

I.e. 
```
<x>1</x>
```

and
```
<x>1</x>
<x>2</x>
```

are both XML, and without some length (or other) delimiter you can't say of it's single message with 2 `x` elements, or 2 messages with singe `x` element each.

Reply:
Ah, I think I'm starting to understand -- the `WriteBufferToKafkaProducer` looked like it might write `rows_per_message` rows per message, so I thought it was using fixed length to delimit them, but in practice it is hard-coded to 1...

Reply:
Yeah, it's misleading...